pulsar-client-cpp/lib/PartitionedConsumerImpl.cc

```diff
@@ -542,11 +542,27 @@ void PartitionedConsumerImpl::handleGetConsumerStats(Result res, BrokerConsumerS
 }
 
 void PartitionedConsumerImpl::seekAsync(const MessageId& msgId, ResultCallback callback) {
-    callback(ResultOperationNotSupported);
+    Lock stateLock(mutex_);
+    if (state_ != Ready) {
+        callback(ResultAlreadyClosed);
+        return;
+    }
+    stateLock.unlock();
+    for (ConsumerList::const_iterator i = consumers_.begin(); i != consumers_.end(); i++) {
+        (*i)->seekAsync(msgId, callback);
```

Comment:
The message id is only relative to 1 partition. Seeking with same message id on multiple partition will position the subscription in the wrong position.

Reply:
I understood that.
This PR decides to support only seeking on partitioned topic by time.